Various updates

ids1024 opened the pull request:
* Remove `TODO` rows from dropdowns
* Add toggles for showing mounted drives in dock
* Allow setting autohide or intellihide for dock (or showing it all the time)
* Toggle for disabling hot corner instead of multiple options, for now
* Remove "Top of Screen" and "Bottom of Screen" options for workspace picker (still need to be implemented)

Further explanation of each change in commit messages.

ids1024 committed:
Remove "TODO" rows from dropdowns

I believe the plan here was previously to allow selecting specific
monitors. But that is potentially complicated (monitor numbers may not
match intuition, multiple can have same name, hotplugging can change
things...). And it's probably not hugely useful.

We can consider adding that sort of thing later.

ids1024 committed:
Add toggle to show mounts in volume

As per: https://github.com/pop-os/gnome-shell-extension-ubuntu-dock/issues/10

It seems good to have a toggle for this whether or not we make it the
default.

ids1024 committed:
Allow setting dock to autohide, intellihide, or show all the time

Fixes https://github.com/pop-os/gnome-control-center/issues/170

Since `dash-to-dock` supports both autohide and intellihide, might as
well support both.

ids1024 committed:
Use just toggle for enabling/disabling hot corner

We may offer more options in the future.

ids1024 committed:
Remove "Top of Screen" and "Bottom of Screen" as options for workspaces

As has been discussed. As it's implemented in Gnome Shell, supporting
either vertical or horizontal orientation is likely complicated.

This still needs to be implemented in Cosmic to work.

ids1024 committed:
Bind workspace picker side radio buttons to gsettings

Requires https://github.com/pop-os/cosmic/pull/77

ids1024 commented:
Added a commit make the workspace switcher side setting work, using https://github.com/pop-os/cosmic/pull/77.

jacobgkau commented:
The option for the hot corner is currently labeled `Enable top-left hot corner`. Should the option mention `Workspaces` since that's what the hot corner will open? (In the default layout, the `Workspaces` button is in the top-left, but if I disable the `Workspaces` button, then I have an `Applications` button in the top-left and the hot corner still opens `Workspaces`-- I'm just thinking that with the current wording, it might not be intuitive for someone to know the hot corner can't be configured further in a situation like that.)

This also seems to remove the entire `Placement of the Workspace Picker` section. I see that you only removed two of the options, but it's no longer showing up at all:

![Screenshot from 2021-06-18 13-45-32](https://user-images.githubusercontent.com/7199422/122570690-3953ac00-d009-11eb-8e19-8bd8ff58c058.png)

We still want left/right as options, right?

ids1024 commented:
> The option for the hot corner is currently labeled Enable top-left hot corner. Should the option mention Workspaces since that's what the hot corner will open?

I wasn't sure about the best wording for that. "Enable top-left workspaces hot corner"? Hm...

> This also seems to remove the entire Placement of the Workspace Picker section. I see that you only removed two of the options, but it's no longer showing up at all:

The section only shows up if it finds the relevant gsettings key, which requires https://github.com/pop-os/cosmic/pull/77. If you install that branch it should show up and work.

jacobgkau commented:
> The section only shows up if it finds the relevant gsettings key, which requires pop-os/cosmic#77. If you install that branch it should show up and work.

Got it, that's working fine.

> I wasn't sure about the best wording for that. "Enable top-left workspaces hot corner"? Hm...

I was thinking `Enable top-left hot corner for Workspaces`, keeping the "hot corner" term generic but just making clear that it can only be used for Workspaces at this time. Might not hurt to confirm with @maria-komarova or others, whether a change is needed at all and what it should be.

maria-komarova commented:
For the hot corner, I guess the option with the hot corner being disabled is still the default, right? So could we just omit Applications row and have the following section under Desktop > General? Or am I missing some context?
![image](https://user-images.githubusercontent.com/57502897/122593309-b3445f00-d022-11eb-96d9-8c8b8aab4020.png)

jackpot51 commented:
Yes, @maria-komarova, that would still be possible.

maria-komarova commented:
Sorry, missed that we were trying to make it a toggle. Wonder if bullet points could be more clear since we plan to add Applications eventually.

ids1024 commented:
Currently in this branch it looks like this (with the wording @jacobgkau suggested).

![Screenshot from 2021-06-18 09-51-33](https://user-images.githubusercontent.com/2263150/122593638-cb17e500-d01a-11eb-9876-305644390adf.png)

> Wonder if bullet points could be more clear since we plan to add Applications eventually.

Seems a bit strange to have radio buttons for something that's just enabled or disabled. It could be changed to radio buttons later when more options are added.

maria-komarova commented:
Sounds good. Yes, you are right, toggle seems better. I wonder if we should put it under Workspaces for now instead of General since it is only for Workspaces. 
Would the following copy work a bit better? "Open Workspaces using Top-left Hot Corner"

WatchMkr commented:
Looking forward, General is probably better when we add Applications and different functionality per corner.

WatchMkr commented:
> Allow setting autohide or intellihide for dock (or showing it all the time)

I'd like to change our current description to intellihide and perhaps leave out autohide. 

This bug causes problems with intelihide that I think would alleviate the need for autohide: https://github.com/pop-os/gnome-shell-extension-ubuntu-dock/issues/8

maria-komarova commented:
Sounds good. I believe if we figure out a fix for this bug, the behavior would be improved significantly.

ids1024 commented:
I think some people will still want to just set it to autohide. Does leaving it out as an option really make things much simpler?

WatchMkr commented:
I think it's more about clarity. What does autohide provide that inteilihide doesn't? And if there's an additional similar feature, can we describe the differences so there's little hesitation about which one the user wants?

jacobgkau commented:
The three options listed in [this comment](https://github.com/pop-os/beta/issues/140#issuecomment-854973512) seem to describe the differences well, and can be summarized succinctly:

- Automatic: Dock always hides unless actively being revealed by the mouse.
- Intelligent hiding, active window only: Dock hides only when the active window overlaps the dock area.
- Intelligent hiding, all windows: Dock hides when any window overlaps the dock area, regardless of focus.

Instead of a drop-down, maybe this would work better as a bullet-point list so the descriptions can be included? (With "never" being the first/default option in the list.)

Upstream Dash to Dock seems to have separate on/off settings for Autohide and Intellihide, and has an additional checkbox setting for "application based intellihide": https://micheleg.github.io/dash-to-dock/media/settings1.png 

That upstream panel looks too confusing, but if the features are present anyway (which it [looks like they are](https://github.com/pop-os/gnome-shell-extension-ubuntu-dock/blob/4add3f3c927c0fb133be4b83326ab4ce5f54f1e2/intellihide.js#L255)), being able to configure seems better than not being able to configure, especially for getting former Dash to Dock users comfortable using the built-in fork. The "active window only" mode seems the most useless of the three to me, but I could see someone wanting the dock to always hide for aesthetic purposes.

jacobgkau commented:
Maybe these wordings would work better for bullet points. Hide dock:

- Never: Dock is always visible.
- Behind active window: Dock hides only when the active window overlaps the dock area.
- Behind any window: Dock hides when any window overlaps the dock area, regardless of focus.
- Always: Dock always hides unless actively being revealed by the mouse.

And that's in order of "least hidden" to "most hidden." Just an idea.

maria-komarova commented:
The Active window only seems to be what we currently have - seems to fit with the bug @WatchMkr mentioned above. That behavior was found to be frustrating during the diary study we did. So I would argue against including it. 
Auto-hide could potentially be a desired behavior but if intellihide is done right (behind any window) then the only difference is if someone wants to see the empty display without a dock on it. Are there many use cases when this is useful?

WatchMkr commented:
I like where you're going. Application based is a weird one though. It conflicts with tiling layouts and probably isn't useful enough to include. I'm hoping that's what we have set and we don't actually have a Dash to Dock bug to hunt down.

How about:

**Dock Visibility**
Always visible
Always hide: Dock always hides unless actively being revealed by the mouse.
Intelligently hide: Dock hides when any window overlaps the dock area.

maria-komarova commented:
Sounds good.

ids1024 committed:
Implement "Show * Icon in Dock" settings

Rather complicated, but the binding works in both directions and
preserves order.

ids1024 commented:
Updated to use radio buttons:

![Screenshot from 2021-06-18 15-10-03](https://user-images.githubusercontent.com/2263150/122620275-6030d300-d047-11eb-80f1-4ae442a27182.png)

Also added a commit implementing the "Show * Icon in Dock" settings. With that, everything here should now be implemented.
